packages/flow-core/src/components/f-suggest/f-suggest.ts

```diff
@@ -271,29 +414,49 @@ export class FSuggest extends FRoot {
 	}
 
 	getSuggestionHtml(suggestions: FSuggestSuggestions) {
-		if (this.isSuggestionArray) {
+		if (this.isSuggestionArray && !this.isSuggestionObjects) {
 			if (this.anySuggestions) {
 				return html`<f-div height="hug-content" direction="column"
-					>${(suggestions as string[]).map(sg => {
+					>${(suggestions as string[]).map((sg, index) => {
 						return html`<f-div
 							class="f-select-options-clickable"
 							height="hug-content"
 							@click=${this.handleSuggest}
 							clickable
 							padding="medium"
+							.selected=${index === this.currentIndex ? "background" : "none"}
 						>
 							<f-div direction="row" gap="medium">
 								${this.isSearchComponent ? html` <f-icon source="i-search"></f-icon>` : ""}
-								<f-text variant="para" size="small" weight="regular"> ${unsafeHTML(sg)} </f-text>
+								<f-text variant="para" size="small" weight="regular" .highlight=${this.value}>
+									${unsafeHTML(sg)}
+								</f-text>
 							</f-div>
 						</f-div>`;
 					})}</f-div
 				>`;
 			}
 			return nothing;
+		} else if (this.isSuggestionObjects) {
+			if (this.anySuggestions) {
+				return html`<f-div height="hug-content" direction="column"
+					>${(suggestions as FSuggestTemplate[]).map((sg, index) => {
+						return html`<f-div
+							class="f-select-options-clickable"
+							height="hug-content"
+							@click=${() => this.handleSelect(sg)}
+							clickable
+							.selected=${index === this.currentIndex ? "background" : "none"}
+						>
+							${unsafeHTML(getComputedHTML(sg.template(this.value)))}
+						</f-div>`;
+					})}</f-div
+				>`;
+			}
+			return nothing;
 		} else {
```

Comment:
this else condition will never execute here
you need 3 computed properties

isStringArraySuggestions
isCategorySuggestions
isTemplateArraySuggestions